test/unit/processing/due-dates/get-payment-schedule.test.js

```diff
@@ -0,0 +1,34 @@
+jest.mock('../../../../app/processing/due-dates/get-schedule')
+const { getSchedule: mockGetSchedule } = require('../../../../app/processing/due-dates/get-schedule')
+
+const { DUE_DATE } = require('../../../mocks/values/due-date')
+
+const { Q4, M12, T4 } = require('../../../../app/constants/schedules')
+const { MONTH, DAY } = require('../../../../app/constants/time-periods')
+
+const { getPaymentSchedule } = require('../../../../app/processing/due-dates/get-payment-schedule')
+
+const settledValue = 100
+const totalValue = 1000
+const currentDate = new Date()
+
+describe('get payment schedule', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should calculate quarterly schedule for quarterly payments', () => {
+    getPaymentSchedule(Q4, DUE_DATE, settledValue, totalValue, currentDate)
+    expect(mockGetSchedule).toHaveBeenCalledWith(expect.any(Object), 4, settledValue, totalValue, 3, MONTH, currentDate)
+  })
+
+  test('should calculate monthly schedule for monthly payments', () => {
+    getPaymentSchedule(M12, DUE_DATE, settledValue, totalValue, currentDate)
+    expect(mockGetSchedule).toHaveBeenCalledWith(expect.any(Object), 12, settledValue, totalValue, 1, MONTH, currentDate)
+  })
+
+  test('should calculate test quarterly schedule for test quarterly payments', () => {
```

Comment:
For testing quarterly payments, the T4 code is used instead of Q4.

That instruction is split every payment into four, three days apart instead of three months.

Means for end to end testing, no need for teams to wait three months to see the results of the scheduling downstream.

You won't see any T4 codes in production, but used in test occassionally.

Reply:
Haha, yeah that makes sense. Thanks